console: Add configurable newline command and RETURN key behavior

vmk94 opened the pull request:
To address compatibility issues with popular serial console applications like PuTTY, this commit introduces a configurable parameter in the source code. Previously, the console code used '\n' for newline output and expected '\n' as input to move to the beginning of a new line. However, PuTTY sends '\r' (Carriage Return) when the RETURN key is pressed and expects "\r\n" to move the cursor to the beginning of a new line. With this update, the newline command and the behavior of the RETURN key can be customized to ensure compatibility with PuTTY and similar applications.

vmk94 committed:
console: Add configurable newline command and RETURN key behavior

To address compatibility issues with popular serial console applications
like PuTTY, this commit introduces a configurable parameter in the source
code. Previously, the console code used '\n' for newline output and
expected '\n' as input to move to the beginning of a new line. However,
PuTTY sends '\r' (Carriage Return) when the RETURN key is pressed and
expects "\r\n" to move the cursor to the beginning of a new line. With
this update, the newline command and the behavior of the RETURN key can
be customized to ensure compatibility with PuTTY and similar applications.

bgomberg reviewed (changes requested):
Thanks for the PR and my apologies for the delay in getting to this.

I'd be happy to merge this, but would want to keep the default configuration matching the current behavior as to not break any existing users. Could you please update the default values in `console_config.h` accordingly? Also, if you could add a short blurb documenting these new options to the "Compile Options" section of `console/README.md` that would be much appreciated.

vmk94 committed:
Merge branch 'bgomberg:master' into console_support

vmk94 committed:
console: update the configurable macros to its default value

In order to maintain compatibility, the values of newly
introduced MACROS `CONSOLE_RETURN_KEY` and `CONSOLE_NEWLINE` are
reverted to match the current behavior.

vmk94 committed:
Merge branch 'console_support' of github.com:vmk94/anchor into HEAD

vmk94 commented:
Thank you for your comments. I made the changes as requested.

bgomberg commented:
Merged. Thanks!
